### PR TITLE
Refactored admin backup saving. No longer at round end, more data backed up

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Admin verified connections now cache all verified connections for all admins. (Rather then just the last connection data of the currently connected admins)

Sync with the db now happens at admin load time, not at round end. (this was causing annoyances because servers with long rounds could override the admin db with old/stale data overwritting the fresher data that was written by a server with a shorter round)

Fix backup verification not working if the db thinks it still connected but its not actually still connected.

@Mothblocks @Jordie0608

### CODE REVIEW

File: code/modules/admin/admin_ranks.dm [hunk 1]
```diff
@@ -302,3 +297,103 @@ GLOBAL_PROTECT(protected_ranks)
...
+
+/proc/sync_ranks_with_db()
+	set waitfor = FALSE
+
+	if(IsAdminAdvancedProcCall())
+		to_chat(usr, "<span class='admin prefix'>Admin rank DB Sync blocked: Advanced ProcCall detected.</span>", confidential = TRUE)
+		return
+
+	var/list/sql_ranks = list()
+	for(var/datum/admin_rank/R in GLOB.protected_ranks)
```
Comment (ZephyrTFA) [inline code/modules/admin/admin_ranks.dm:310]:
```suggestion
	for(var/datum/admin_rank/R as anything in GLOB.protected_ranks)
```

File: code/modules/admin/admin_ranks.dm [hunk 2]
```diff
@@ -302,3 +297,103 @@ GLOBAL_PROTECT(protected_ranks)
...
+		sql_ranks += list(list("rank" = R.name, "flags" = R.include_rights, "exclude_flags" = R.exclude_rights, "can_edit_flags" = R.can_edit_rights))
+	SSdbcore.MassInsert(format_table_name("admin_ranks"), sql_ranks, duplicate_key = TRUE)
+	update_everything_flag_in_db()
+
+
+/proc/update_everything_flag_in_db()
+	for(var/datum/admin_rank/R in GLOB.admin_ranks)
```
Comment (ZephyrTFA) [inline code/modules/admin/admin_ranks.dm:317]:
```suggestion
	for(var/datum/admin_rank/R as anything in GLOB.admin_ranks)
```

File: code/modules/admin/admin_ranks.dm (hunk 1 above)
Comment (ZephyrTFA) [reply code/modules/admin/admin_ranks.dm:310]: also can you change these to not be single letter var names

File: code/modules/admin/admin_ranks.dm [hunk 3]
```diff
@@ -302,3 +297,103 @@ GLOBAL_PROTECT(protected_ranks)
...
+		to_chat(usr, "<span class='admin prefix'>Admin rank DB Sync blocked: Advanced ProcCall detected.</span>")
+		return
+
+	if(CONFIG_GET(flag/admin_legacy_system) || !SSdbcore.IsConnected()) //we're already using legacy system so there's nothing to save
+		return
+	sync_ranks_with_db()
+	var/list/sql_admins = list()
+	for(var/i in GLOB.protected_admins)
+		var/datum/admins/A = GLOB.protected_admins[i]
+		sql_admins += list(list("ckey" = A.target, "rank" = A.rank_names()))
```
Comment (ZephyrTFA) [inline code/modules/admin/admin_ranks.dm:359]:
```suggestion
	for(var/holder_ckey in GLOB.protected_admins)
		var/datum/admins/holder = GLOB.protected_admins[holder_ckey]
		sql_admins += list(list("ckey" = holder.target, "rank" = holder.rank_names()))
```

File: code/modules/admin/admin_ranks.dm [hunk 4]
```diff
@@ -302,3 +297,103 @@ GLOBAL_PROTECT(protected_ranks)
...
+	if(CONFIG_GET(flag/admin_legacy_system)) //we're already using legacy system so there's nothing to save
+		return
+
+	//json format backup file generation stored per server
+	var/json_file = file("data/admins_backup.json")
+	var/list/file_data = list(
+		"ranks" = list(),
+		"admins" = list()
+	)
+	for(var/datum/admin_rank/R in GLOB.admin_ranks)
```
Comment (ZephyrTFA) [inline code/modules/admin/admin_ranks.dm:380]:
```suggestion
	for(var/datum/admin_rank/R as anything in GLOB.admin_ranks)
```

**Review by ZephyrTFA — DISMISSED** [review]

**Comment by san7890** [comment]
Would this solve #59823 btw

**Comment by Jordie0608** [comment]
rip

**Pull request merged**